### PR TITLE
Misc HTML doc generation changes

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:

- Use common source URL
- Move doc generation config into function
- Add Changelog link to Hex module info page
- Remove overview page and set readme as default page to prevent duplicated content
- Badges and more badges!
- Work wrapping and markdown cleanup
- Fix typos

### CODE REVIEW

**Comment by bmuller** [comment]
Thanks for the help!  Appreciate a number of these changes!

RE newlines - seems like a number were thrown in, and I'm not sure most (if not all) are necessary.  For instance, newlines after markdown headers, in the README for longer lines, etc.  Are these necessary?

**Comment by kianmeng** [comment]
> Thanks for the help! Appreciate a number of these changes!
> 
> RE newlines - seems like a number were thrown in, and I'm not sure most (if not all) are necessary. For instance, newlines after markdown headers, in the README for longer lines, etc. Are these necessary?

It this what you meant, see screenshot below? Line wrapping?

image

If so, it's just my editor setting as I'm using bigger font at console and have auto line wrapping at 80 chars. 
No biggie, I can revert these changes if you like.

**Comment by bmuller** [comment]
> If so, it's just my editor setting as I'm using bigger font at console and have auto line wrapping at 80 chars.
> No biggie, I can revert these changes if you like.

Yep, exactly.  If you revert the newlines, I think everything else looks great.  Thanks for the help!

**Comment by kianmeng** [comment]
> > If so, it's just my editor setting as I'm using bigger font at console and have auto line wrapping at 80 chars.
> > No biggie, I can revert these changes if you like.
> 
> Yep, exactly. If you revert the newlines, I think everything else looks great. Thanks for the help!

Done. Let me know if there are any other changes.

File: README.md [hunk 1]
```diff
@@ -9,10 +11,9 @@ Endon is an Elixir library that provides helper functions for [Ecto](https://hex
 Yes, Ecto is great!  But there are a few things that are really annoying, and a little syntactic sugar can go a long way.  See the [Features](guides/features.md) page for more info.
 
 #### What does Endon mean?
-"Ecto" is a prefix from Greek έκτός (ektós) meaning "outside".  The opposite of Ecto is "Endon" (Greek ἔνδον) means "within, inner, absorbing, or containing".  Why the opposite of Ecto?  No good reason at all; naming is hard.
+"Ecto" is a prefix from Greek έκτός (ektós) meaning "outside".  The opposite of Ecto is "Endon" (Greek ἔνδον) means "within, inner, absorbing, or containing". Why the opposite of Ecto?  No good reason at all; naming is hard.
```
Comment (bmuller) [inline README.md]: I think this change is also superfluous - I used double spaces between sentences throughout so for consistency I think the double space here can be kept.

File: guides/features.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Features
 
 ## Schema Queries
-By setting an application-wide (or schema module specific) `Ecto.Repo` variable, your code doesn't have to mention the repo over and over (see [here](overview.html#configuration) for setup instructions).
+By setting an application-wide (or schema module specific) `Ecto.Repo` variable, your code doesn't have to mention the repo over and over (see [here](../README.md#configuration) for setup instructions).
```
Comment (bmuller) [inline guides/features.md:4]: I think this will break the link - it should go to overview.html

File: README.md (hunk 1 above)
Comment (bmuller) [reply README.md]: I think this is the only outstanding issue - everything else looks good

Comment (kianmeng) [reply README.md]: > I think this is the only outstanding issue - everything else looks good

Done.

**Comment by bmuller** [comment]
Thanks for the help @kianmeng!

**Pull request merged**